### DISCUSSION TITLE
Support Fluent Bit as a log forwarder

### DISCUSSION THREAD

**Discussion by stevehipwell** [kube-logging/logging-operator#1546] [Ideas]
**Is your feature request related to a problem? Please describe.**
I'm interested in if the operator could replace our current Helm based deployment of collector -> forwarder architecture; but to do this we'd need Fluent Bit as the forwarder.

**Describe the solution you'd like**
I'd like to have the option to use Fluent Bit as the log forwarder.

**Describe alternatives you've considered**
N/A

**Additional context**
We've been running the collector -> forwarder architecture in production for a number of years and the benefits of Fluent Bit over Fluentd are significant.

I'm very familiar with this pattern (I maintain the Fluent Helm charts and the Fluentd Aggregator image) and would be interested in contributing if that helps.

**Comment by pepov** [comment]
This is coming up every now and then, but there are a few reasons against it:
- label based routing is only available in fluentd, the tag based matching is not flexible enough to implement the "flow" model
- syslog-ng is already implemented as a more performant and more flexible aggregator alternative with a growing number of outputs supported
- there are plans to implement a new model which would supersede the current design and that will be the focus for the development team for the next few months

**Reply by sebastian-racs-itk** [reply]
According to the FluentBit project, FluentBit can now do the same and more than FluentD and they see no reason to continue using FluentD. Should it therefore be replaced now? Or maybe make syslog-ng the default?
